### ISSUE TITLE
Invalid java SDK path in Editor Settings.Missing 'bin' directory!

### ISSUE THREAD

**Issue by trash-max** [godotengine/godot#90831]
### Tested versions

- Reproducible in Godot 4.2.2 (Linux)
- Not reproducible in Goodot 4.2.1 (Linux)

### System information

Ubuntu 23.10

### Issue description

Error "Invalid java SDK path in Editor Settings.Missing 'bin' directory!" with exporting game to Android.Error "Invalid java SDK path in Editor Settings.Missing 'bin' directory!" with exporting game to Android, but all java paths a same as Godot 4.2.1 and have no problem to export.

Problem only with 4.2.2 version.

### Steps to reproduce

1. Download Godot 4.2.2 for Linux
2. Try export game to Android

### Minimal reproduction project (MRP)

![2024-04-17 22-15-03](https://github.com/godotengine/godot/assets/63397434/b38a76c9-18f8-4872-8fc5-f0a0fd152a17)
![2024-04-17 22-14-26](https://github.com/godotengine/godot/assets/63397434/e2010a6e-dfa5-49ed-8987-5cc6b5b3e304)

**Comment by trash-max** [comment]
I'll add it. 
In my case, for version 4.2.2, the path should look like this:

/usr/lib/jvm/java-17-openjdk-amd64

i.e. "/bin" should be removed. 
This will not be the default value in the system paths and needs to be done manually.

**Comment by SimoneStarace** [comment]
@trash-max Thank you very much.
I had the same issue but putting this path `/usr/lib/jvm/java-17-openjdk-amd64` solved the problem.

**Comment by BraveEvidence** [comment]
For macOS , run this in terminal `/usr/libexec/java_home`

**Comment by SANYAM591** [comment]
/usr/lib/jvm/java-17-openjdk-amd64 
this is not working for me can someone tell the sollution ?

**Comment by SimoneStarace** [comment]
> /usr/lib/jvm/java-17-openjdk-amd64 this is not working for me can someone tell the sollution ?

Execute this command to find the path of your JDK
`readlink -f $(which java)`

**Comment by damian-c** [comment]
Confirmed that my JDK path was correct, yet this error still persists on the 4.2.2 snap version. This is almost certainly a snap problem, because I was able to get around it by installing the 4.2.2 binary from godotengine.org instead.

**Comment by SimoneStarace** [comment]
Hi @damian-c 

> Confirmed that my JDK path was correct, yet this error still persists on the 4.2.2 snap version. This is almost certainly a snap problem, because I was able to get around it by installing the 4.2.2 binary from godotengine.org instead.

I cannot confirm this since I have installed Godot by downloading the binary file directly from the website.

**Comment by Warzmik** [comment]
Hello, in my case the route in Windows 11 was `C:\Program Files\Eclipse Adoptium\jdk-17.0.10.7-hotspot`

**Comment by devchenli** [comment]
This is bug !!!!!

**Comment by tschwaerzl** [comment]
I had the same issue with the Steam version of Godot `4.3.stable`. Installing it natively 'fixes' it and everything works. Same settings/config on both versions.

**Steam:**
![Image](https://github.com/user-attachments/assets/2ec77a85-566b-4ef4-b7e2-f55e2d055aa3)
![Image](https://github.com/user-attachments/assets/a1c530d8-2092-4459-b9f0-1e733d0bbefd)

**Comment by Alex2782** [comment]
@tschwaerzl 
On my MacOS: `/opt/homebrew/opt/openjdk@17/`
Can you run “java” in “bin” folder?

```
/opt/homebrew/opt/openjdk@17/bin/java --version

openjdk 17.0.8.1 2023-08-24
OpenJDK Runtime Environment Homebrew (build 17.0.8.1+0)
OpenJDK 64-Bit Server VM Homebrew (build 17.0.8.1+0, mixed mode, sharing)
```

**Comment by tschwaerzl** [comment]
Sure and yes.
```bash
➜  game git:(main) /usr/lib/jvm/java-17-openjdk/bin/java --version
openjdk 17.0.13 2024-10-15
OpenJDK Runtime Environment (build 17.0.13+11)
OpenJDK 64-Bit Server VM (build 17.0.13+11, mixed mode, sharing)
```

**Comment by Alex2782** [comment]
What happens if you run `chmod +777` on the 'bin' folder? (give all rights, including writing rights)

Godot just tries to open the folder:

```cpp
		// Check for the bin directory.
		Ref<DirAccess> da = DirAccess::open(java_sdk_path.path_join("bin"), &errn);
		if (errn != OK) {
			err += TTR("Invalid Java SDK path in Editor Settings.");
			err += TTR("Missing 'bin' directory!");
			err += "\n";
			valid = false;
		} 
```

**Comment by tschwaerzl** [comment]
Moin @Alex2782; with full access rights (777) set on the Java SDK `bin` folder I receive the same error. If you're on Discord we can do a live debug session if this helps.

As mentioned, it's only the Steam version (`v4.3.stable.official [77dcf97d8]`) which throws the error. Launching Godot from the Arch ([pacman](https://archlinux.org/packages/extra/x86_64/godot/)) installation (`v4.3.stable.arch_linux`) everything works fine.

**Comment by Lazres31** [comment]
if you really need the steam version, you can download [java](https://www.oracle.com/java/technologies/downloads/) from the official site, unpack zip into Download folder and set path to it. It works for me

**Comment by binghamfluid** [comment]
There seams to be a access right problem for the steam version of godot to get access to the /usr/lib/jvm/jdk-version
Copying all files from /jvm/jdk-version to a path like $USERHOME/jdk-version and point godot to this path solves this problem.

**Comment by SOSeriously** [comment]
If you are using linux and have updated your system. Just make sure that open JDK 17 was not deleted in the process. It happened to me. I just had to install Open JDK 17 again and change the path in Godot.

**Comment by binghamfluid** [comment]
> If you are using linux and have updated your system. Just make sure that open JDK 17 was not deleted in the process. It happened to me. I just had to install Open JDK 17 again and change the path in Godot.

nope, that is not the problem here. You can try it by yourself. Install godot with steam and outside of steam. Config both the same and the steam-version will fail to make a android version.

**Comment by timvisee** [comment]
I second this, and am seeing the same.

The last few Steam versions fail on the invalid SDK path. The non-Steam version from the website works totally fine.

I'd be happy to debug if there are any specific questions.

**Comment by biruk10hana-sys** [comment]
I developed my first game in godot but when I tried to export it invalid jdk showed up. I am using v4.4.1. On windows 11. I already checked the path more than once. How can I fix it

**Comment by Rindbee** [comment]
Probably because the editor is in **sandbox** mode. At this time, access to certain directories is restricted.

You can bypass Steam and access the binary directly. Access directly from the command line, or create a desktop shortcut to the binary.

In Linux Mint 22, the default path for Godot in Steam is:

```
~/.steam/debian-installation/steamapps/common/Godot\ Engine/godot.x11.opt.tools.64
```

Alternatively, copy the JDK directory to an accessible location (such as your home directory or /opt/).

**Comment by binghamfluid** [comment]
> Probably because the editor is in **sandbox** mode. At this time, access to certain directories is restricted.

Ok. Then we should update the documentation with this as a hint or make a check inside godot if the selected path can be reached. If not we should open a warning to the user.

**Comment by N3COL** [comment]
<img width="1541" height="884" alt="Image" src="https://github.com/user-attachments/assets/eb66f676-c81e-45b6-9f56-47e6ca34c831" />

<img width="1220" height="726" alt="Image" src="https://github.com/user-attachments/assets/bc5e6707-a35d-49fd-bc89-f3aca42728b0" />

[Ubuntu 24.04] [Godot 4.6 from Steam]
I had that problem and  solved this that:

1. Install Android Studio (Panda 3)
2. And because this file have already java so i pasted path of jbr from extracted file (of Android St.) 

This looks like this now:
`[path above file]/android-studio-panda3-patch1-linux/android-studio/jbr`

<img width="1277" height="216" alt="Image" src="https://github.com/user-attachments/assets/2752f34c-3c5c-4525-a810-942dd72e2af5" />
And I think its work now

<img width="1277" height="734" alt="Image" src="https://github.com/user-attachments/assets/9eba6b7d-df27-4cb9-9052-ede15cb1280c" />
(The yellow error is only about name project so dont care)``
